src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/builder/BuiltCondition.java

```diff
@@ -0,0 +1,57 @@
+package io.stargate.sgv2.jsonapi.service.operation.model.impl.builder;
+
+// TODO: this is a bit of a mess, it is actually building the query string, prob from the copied old
+// code
+public final class BuiltCondition {
+
+  public ConditionLHS lhs;
+
+  public BuiltConditionPredicate predicate;
+
+  public BuildConditionTerm rhsTerm;
+
+  public BuiltCondition(
+      ConditionLHS lhs, BuiltConditionPredicate predicate, BuildConditionTerm rhsTerm) {
+    this.lhs = lhs;
+    this.predicate = predicate;
+    this.rhsTerm = rhsTerm;
+  }
+
+  public static BuiltCondition of(
+      ConditionLHS lhs, BuiltConditionPredicate predicate, BuildConditionTerm rhsTerm) {
+    return new BuiltCondition(lhs, predicate, rhsTerm);
+  }
+
+  public static BuiltCondition of(
+      String columnName, BuiltConditionPredicate predicate, BuildConditionTerm rhsTerm) {
+    return of(ConditionLHS.column(columnName), predicate, rhsTerm);
+  }
+
+  @Override
+  // TODO Clarify if this is building the CQL string or just the toString for debugging
+  // I think it is jusr debugging becaus the query is build in the QueryBuilder
```

Comment:
So not an ideal, code duplication -- but as you suggested, I think it's copied code.

Reply:
Yes, this is basically what is in quarkus-common originally.

Reply:
Yuqi Resolve: this comment can be resolved now, there is a TODO here